### PR TITLE
ablative hood head reflects now

### LINKED ISSUE #47625: Reflector trenchcoat hood does not reflect energy projectiles
Replicate by equipping reflector trenchcoat and then toggling the hood, followed by aiming at the head. 0% reflect chance but it absorbs energy damage, probably not intended considering aiming anywhere else will have a chance to reflect.

### PR DESCRIPTION
**Pull request by ghost**
closes #47625
:cl:
fix: Ablative trenchcoat's hood reflects now.
/:cl:

### CODE REVIEW

**Comment by carlarctg** [comment]
what about the arms n legs

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 1]
```diff
@@ -89,12 +89,15 @@
 
 /mob/living/carbon/human/proc/check_reflect(def_zone) //Reflection checks for anything in your l_hand, r_hand, or wear_suit based on the reflection chance of the object
 	if(wear_suit)
-		if(wear_suit.IsReflect(def_zone) == 1)
-			return 1
+		if(wear_suit.IsReflect(def_zone) == TRUE)
```
Comment (ninjanomnom) [inline code/modules/mob/living/carbon/human/human_defense.dm]:
```suggestion
		if(wear_suit.IsReflect(def_zone))
```

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 2]
```diff
@@ -89,12 +89,15 @@
 
 /mob/living/carbon/human/proc/check_reflect(def_zone) //Reflection checks for anything in your l_hand, r_hand, or wear_suit based on the reflection chance of the object
 	if(wear_suit)
-		if(wear_suit.IsReflect(def_zone) == 1)
-			return 1
+		if(wear_suit.IsReflect(def_zone) == TRUE)
+			return TRUE
+	if(head)
+		if(head.IsReflect(def_zone) == TRUE)
```
Comment (ninjanomnom) [inline code/modules/mob/living/carbon/human/human_defense.dm]:
```suggestion
		if(head.IsReflect(def_zone))
```

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 3]
```diff
@@ -89,12 +89,15 @@
...
+		if(wear_suit.IsReflect(def_zone) == TRUE)
+			return TRUE
+	if(head)
+		if(head.IsReflect(def_zone) == TRUE)
+			return TRUE
 	for(var/obj/item/I in held_items)
-		if(I.IsReflect(def_zone) == 1)
-			return 1
-	return 0
+		if(I.IsReflect(def_zone) == TRUE)
```
Comment (ninjanomnom) [inline code/modules/mob/living/carbon/human/human_defense.dm]:
```suggestion
		if(I.IsReflect(def_zone))
```

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Comment by optimumtact** [comment]
the account got killed

**Pull request closed without merging**